samcli/commands/init/interactive_init_flow.py

```diff
@@ -149,57 +220,94 @@ def _generate_from_app_template(
         _package_schemas_code(runtime, schemas_api_caller, schema_template_details, output_dir, name, location)
 
 
-def _get_runtime(runtime):
-    if not runtime:
-        choices = list(map(str, range(1, len(INIT_RUNTIMES) + 1)))
-        choice_num = 1
-        click.echo("\nWhich runtime would you like to use?")
-        for r in INIT_RUNTIMES:
-            msg = "\t" + str(choice_num) + " - " + r
-            click.echo(msg)
-            choice_num = choice_num + 1
-        choice = click.prompt("Runtime", type=click.Choice(choices), show_choices=False)
-        runtime = INIT_RUNTIMES[int(choice) - 1]  # zero index
-    return runtime
+def _get_app_template_choice(templates_options, dependency_manager):
+    templates = _get_templates_with_dependency_manager(templates_options, dependency_manager)
+    chosen_template = templates[0]
+    if len(templates) > 1:
+        click.echo("\nSelect your starter template")
+        click_template_choices = []
+        for idx, template in enumerate(templates):
+            click.echo("\t{index} - {name}".format(index=idx + 1, name=template["displayName"]))
+            click_template_choices.append(str(idx + 1))
+        template_choice = click.prompt("Template", type=click.Choice(click_template_choices), show_choices=False)
+        chosen_template = templates[int(template_choice) - 1]
+    return chosen_template
+
+
+def _get_templates_with_dependency_manager(templates_options, dependency_manager):
+    templates = []
+    for template in templates_options:
+        if template["dependencyManager"] == dependency_manager:
+            templates.append(template)
+    return templates
+
+
+def _get_choice_from_options(choosen, options, question, msg):
+
+    if not choosen:
```

Comment:
You may want to reverse this condition to short circuit the block and reduce indentation:
```
if chosen is not None:
   return chosen
```